### PR TITLE
feat: add support for extra disk management

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

**Comment by andrewrynhard** [comment]
```
[   22.220261] [talos] [initramfs] performing platform specific tasks
[   22.229075] [talos] [initramfs] partitioning /dev/sda - ESP
[   22.236783] [talos] [initramfs] partitioning /dev/sda - ROOT
[   22.244472] [talos] [initramfs] partitioning /dev/sda - DATA
[   22.253250] ata3.00: Enabling discard_zeroes_data
[   22.259369]  sda: sda1 sda2 sda3
[   22.263402] [talos] [initramfs] formatting partition /dev/sda1 - ESP
[   22.291783] [talos] [initramfs] formatting partition /dev/sda2 - ROOT
[   22.337883] [talos] [initramfs] formatting partition /dev/sda3 - DATA
[   22.388723] [talos] [initramfs] partitioning /dev/sdb -
[   22.396718] ata4.00: Enabling discard_zeroes_data
[   22.402826]  sdb: sdb1
[   22.405790] [talos] [initramfs] formatting partition /dev/sdb1 -
[   22.510697] [talos] [initramfs] mounting the partitions
[   22.534111] ata3.00: Enabling discard_zeroes_data
[   22.540505]  sda: sda1 sda2 sda3
[   22.546227] XFS (sda2): Mounting V5 Filesystem
[   22.568487] XFS (sda2): Ending clean mount
[   22.579405] XFS (sda3): Mounting V5 Filesystem
[   22.602853] XFS (sda3): Ending clean mount
[   22.893285] [talos] [initramfs] downloading assets for /root/boot
```

File: internal/pkg/install/prepare.go [hunk 1]
```diff
@@ -312,25 +328,23 @@ type Target struct {
...
 	if pt, err = bd.PartitionTable(true); err != nil {
 		return err
 	}
 
+	opts := []interface{}{partition.WithPartitionTest(t.Test)}
+
 	switch t.Label {
 	case constants.BootPartitionLabel:
 		// EFI System Partition
-		typeID = "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
```
Comment (bradbeam) [inline internal/pkg/install/prepare.go]: Is it worth moving these to const's right now or todo for later?

File: pkg/userdata/userdata.go [hunk 2]
```diff
@@ -121,11 +121,12 @@ type File struct {
...
 type Install struct {
-	Boot  *BootDevice    `yaml:"boot,omitempty"`
-	Root  *RootDevice    `yaml:"root"`
-	Data  *InstallDevice `yaml:"data,omitempty"`
-	Wipe  bool           `yaml:"wipe"`
-	Force bool           `yaml:"force"`
+	Boot         *BootDevice      `yaml:"boot,omitempty"`
+	Root         *RootDevice      `yaml:"root"`
+	Data         *InstallDevice   `yaml:"data,omitempty"`
+	ExtraDevices []*InstallDevice `yaml:"extraDevices,omitempty"`
```
Comment (bradbeam) [inline pkg/userdata/userdata.go]: should we allow mount points to be configured as part of the extra devices?

File: internal/pkg/install/prepare.go (hunk 1 above)
Comment (andrewrynhard) [reply internal/pkg/install/prepare.go]: I think I'd prefer that in another PR.

File: pkg/userdata/userdata.go (hunk 2 above)
Comment (andrewrynhard) [reply pkg/userdata/userdata.go]: Good point. We absolutely should. Otherwise they have no way to handle that. Good catch.

**Comment by andrewrynhard** [comment]
UX:

```
install:
  ...
  extraDevices:
  - device: /dev/sdb
    partitions:
    - size: 40960000000
      mountpoint: /var/lib/app
```

File: internal/app/init/internal/rootfs/mount/mount.go [hunk 3]
```diff
@@ -130,6 +132,24 @@ func (i *Initializer) InitOwned() (err error) {
 	return nil
 }
 
+// ExtraDevices mounts the extra devices.
+func ExtraDevices(data *userdata.UserData) (err error) {
+	if data.Install == nil || data.Install.ExtraDevices == nil {
+		return nil
+	}
+	for _, extra := range data.Install.ExtraDevices {
+		for i, part := range extra.Partitions {
```
Comment (bradbeam) [inline internal/app/init/internal/rootfs/mount/mount.go]: last nit, do we want to throw an error here if the mount point is not under `/var`?

Comment (andrewrynhard) [reply internal/app/init/internal/rootfs/mount/mount.go]: Ahh good point. Should be simple enough 👍.

Comment (andrewrynhard) [reply internal/app/init/internal/rootfs/mount/mount.go]: Gonna make this happen in a followup PR. I would like to get this in so we can use it for our storage.

**Pull request merged**